test/Console/Command/RunCommandTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace GrumPHPTest\Console\Command;
+
+use GrumPHP\Configuration\GrumPHP;
+use GrumPHP\Console\Command\RunCommand;
+use GrumPHP\Locator\RegisteredFiles;
+use PHPUnit\Framework\TestCase;
+
+class RunCommandTest extends TestCase
```

Comment:
This test can be dropped if you change the input option logic as suggested above.

Reply:
I'dont feel quite comfortable having "nothing" that actually checks the behavior - which kinda was the initial thought on adding a test in the first place.

E.g. consider the inlined
```php
 $tasks = array_map('trim', explode(',', $input->getOption('tasks', '')));
```
If somebody were to switch "," by ";" there would be "nothing" to catch that bug. Mutation testing tools like https://github.com/infection/infection usually "make those changes" to verify if everything is tested "correctly".

---

Are you open to introducing some sort of end2end/integration test that "acutally" calls the run command and verifies the output (see https://symfony.com/doc/4.1/console.html#testing-commands )?

It's something that we do quite frequently because it's a very good high level indicator for "some things don't work as they are supposed to" (though it comes with an increased maintenance overhead due to duplication).

Something along the lines of

```php
    public function test_command()
    {
        $application = new Application();

        /**
         * @var RunCommand $command
         */
        $command = $application->find('run');

        $commandTester = new CommandTester($command);
        $commandTester->execute(array(
            'command'  => $command->getName(),
            '--tasks' => 'foo',
        ));

        $output = $commandTester->getDisplay();
        $this->assertContains('...', $output);
    }
```

Reply:
Hi @paslandau,

In a normal situation, those tests would work fine.
However, the behaviour of the grumphp commands are different based on which configuration file you supply. This means we would have a lot of e2e tests to cover all possible situations.
I was looking in to this with @Landerstraeten a while ago, but didn't find a good solution yet.

Feel free to suggest something in this PR.

Reply:
I'll give it some thought but I'm not to deep into symfony (more of a laravel guy) plus since those sort of test are usually tidly coupled to the application I guess I'd need more time digging some grumphp internals.

So I would keep i separate from this PR + keep the current test for now.